test/e2e/versioned_explicit_reference_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+
+	uitest "github.com/cppforlife/go-cli-ui/ui/test"
+)
+
+func TestVersionedExplicitReference(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+
+	yaml1 := `
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-1
+  annotations:
+    kapp.k14s.io/versioned: ""
+data:
+  foo: bar
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-2
+  annotations:
+    kapp.k14s.io/versioned-explicit-ref: |
+      apiVersion: v1
+      kind: ConfigMap
+      name: config-1
+data:
+  foo: bar
```

Comment:
probably worth adding some additional refs that do not match and may be an additional resource with more refs.

Reply:
Will do. Also thinking about adding tests for the key validation since they are [hardcoded keys](https://github.com/vmware-tanzu/carvel-kapp/pull/311/files#diff-3a4768871151150b99289f09a0d3b8a6be2926d600300178104cc94ac894a460R34).

Reply:
This does not seem to be possible today as `RunWithOpts` errors out in case of an error.